server.js

```diff
@@ -3492,29 +3492,52 @@ function mapNugetFeed(pattern, offset, getInfo) {
   var vPreRegex = new RegExp('^\\/' + pattern + '\\/vpre\\/(.*)\\.(svg|png|gif|jpg|json)$');
 
   function getNugetVersion(apiUrl, id, includePre, request, done) {
-    var reqUrl = apiUrl + '/flatcontainer/' + id.toLowerCase() + '/index.json';
-    request(reqUrl, function(err, res, buffer) {
-      if (err != null) {
-        done(err);
-        return;
-      }
-
-      try {
+    // get service index document
+    regularUpdate(apiUrl + '/index.json',
+      (3600 * 1000),
```

Comment:
Another way to think about it is to tolerate a percentage `x` of all NuGet badge requests to fail. The probability of a single request to fail is `(update period) / (endpoint change period)`; the expected value of the number of failing requests is `(number of requests per year) * (update period) / (endpoint change period)` (expectation of a binomial distribution). So the update period is `(expected value of number of failing requests per year) * (endpoint change period) / (number of requests per year)`, ie. `(proportion of failing requests) * (endpoint change period)`. If we take the proportion of failing requests to be `10^-4` (0.01%), it comes out to 52.56 minutes.